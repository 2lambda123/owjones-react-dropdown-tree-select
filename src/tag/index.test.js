import { mount } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import Tag from './index'

const nativeEvent = { nativeEvent: { stopImmediatePropagation: () => {} } }

describe('<Tag/>', () => {
  it('renders label when passed in', () => {
    const tree = renderer
      .create(<Tag label='hello' id='abc' />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('call onDelete handler when pill is closed', () => {
    const onDelete = jest.fn()
    const wrapper = mount(<Tag label='hello' id='abc' onDelete={onDelete} />)
    wrapper.find('.tag-remove').simulate('click', nativeEvent)
    expect(onDelete).toBeCalledWith('abc')
  })

  it('should not cause form submit', () => {
    const onSubmit = jest.fn()
    const onDelete = jest.fn()
    const wrapper = mount(
      <form onSubmit={onSubmit}>
        <Tag label='hello' id='abc' onDelete={onDelete} />
      </form>
    )
    wrapper.find('.tag-remove').simulate('click', nativeEvent)
    expect(onSubmit).not.toBeCalled()
  })
})
