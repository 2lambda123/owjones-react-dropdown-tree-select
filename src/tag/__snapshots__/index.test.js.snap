// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag/> renders label when passed in 1`] = `
<span
  className="tag"
>
  hello
  <button
    className="tag-remove"
    onClick={[Function]}
    type="button"
  >
    x
  </button>
</span>
`;
